setup.py

```diff
@@ -125,129 +125,182 @@ def clean_line(line):
-    platforms="any",
-)
+#%% Main install routine
+def run_setup(with_binary):
+    setup(
+        name="radis",
+        version=__version__,
+        description=description,
+        long_description=long_description,
+        long_description_content_type="text/markdown",
+        url="https://github.com/radis/radis",
+        author="Erwan Pannier",
+        author_email="erwan.pannier@gmail.com",
+        license="GNU Lesser General Public License v3 (LGPLv3)",
+        keywords=[
+            "spectrum",
+            "infrared",
+            "spectra",
+            "radiation",
+            "nonequilibrium",
+            "spectroscopy",
+            "molecules",
+            "HITRAN",
+        ],
+        packages=find_packages(),
+        install_requires=[
+            "hitran-api",
+            "numpy",
+            "scipy>=1.4.0",
+            "matplotlib",
+            "cython",
+            "pandas>=1.0.5",
+            "plotly>=2.5.1",
+            "numba",
+            "mpldatacursor",
+            "astropy",  # Unit aware calculations
+            "publib>=0.3.2",  # Plotting styles for Matplotlib
+            "plotly>=2.5.1",  # for line survey HTML output
+            "termcolor",  # terminal colors
+            "configparser",
+            "astroquery>=0.3.9",  # to fetch HITRAN databases
+            "json-tricks>=3.15.0",  # to deal with non jsonable formats
+            "tables",  # for pandas to HDF5 export
+            "pytest",  # to run test suite
+            "joblib",  # for parallel loading of SpecDatabase
+            "numba",  # just-in-time compiler
+        ],
+        extras_require={
+            "dev": [
+                "numpydoc",  # for Jedi (autocompletion) to recognize
+                "black>=20.8b1",  # for code-linting in accordance to PEP8
+                "isort",  # for sorting imports
+                "pre-commit",  # to enforce Black before each commit
+            ]
+        },
+        classifiers=[
+            "Development Status :: 4 - Beta",
+            "Intended Audience :: Science/Research",
+            "License :: OSI Approved :: GNU Lesser General Public License v3 (LGPLv3)",
+            "Topic :: Scientific/Engineering",
+            "Programming Language :: Python",
+            "Programming Language :: Python :: 3.4",
+            "Programming Language :: Python :: 3.5",
+            "Programming Language :: Python :: 3.6",
+            "Programming Language :: Python :: 3.7",
+            "Operating System :: OS Independent",
+        ],
+        **get_ext_modules(with_binary),
+        include_package_data=True,  # add non .py data files in MANIFEST.in
+        # package_data={'radis': ['radis/phys/units.txt']},
+        zip_safe=False,  # impossible as long as we have external files read with __file__ syntax
+        platforms="any",
+    )
+
+
+# %% Run Main install routine
+try:
+    run_setup(with_binary=True)
+except BuildFailed:
+    show_message(
```

Comment:
It should also print the reason why `run_setup(with_binary=True)` failed.

Reply:
done !